WMIAdventure/frontend/src/js/components/profile/molecules/UpgradeCardSection/UpgradeCardSection.js

```diff
@@ -23,31 +23,50 @@ class UpgradeCardSection extends React.Component {
             translateY: '-100vh',
         },
         cardLevels: null,
+        cardDescriptions: null,
         approvePopUpHover: true,
     }
 
     componentDidMount() {
-        this.loadCardLevels();
+        this.loadCardLevelsAndDescriptions();
     }
 
-    async loadCardLevels() {
+    async loadCardLevelsAndDescriptions() {
         const card = await getCardById(this.props.cardId);
         let levels = [];
-        for (let elem of card.levels)
+        let descriptions = [];
+        for (let elem of card.levels) {
             levels.push(elem.level);
+            descriptions.push(elem.effects_description);
+        }
         this.setState({
-            cardLevels: levels
+            cardLevels: levels,
+            cardDescriptions: descriptions
         });
     }
 
-    getNextCardLevel = () => {
-        if (this.state.cardLevels === null) return;
-        if (this.props.cardLevel === 1)
-            // case [1, 2] and [1, 3]
-            return this.state.cardLevels[1];
-        else if (this.props.cardLevel === 2)
+    getNextCardData = (data) => {
+        if (data === null) return;
+        if (data.length === 2)
+            // case [1, 2], [2, 3] and [1, 3]
+            return data[1];
+        else if (data.length === 3) {
             // case [1, 2, 3]
-            return this.state.cardLevels[2];
+            if (this.props.cardLevel === 1)
+                // when typical return gold
+                return data[1];
+            else if (this.props.cardLevel === 2)
+                // when gold return epic
+                return data[2];
+        }
+    }
+
+    getNextCardLevel = () => {
+        return this.getNextCardData(this.state.cardLevels);
+    }
+
+    getNextCardDescription = () => {
+        return this.getNextCardData(this.state.cardDescriptions);

```

Comment:
Może lepiej zwrócić pusty string `' '` jeżeli nie uda się pobrać tych danych? Może to powodować jakieś problemy więc lepiej to zrobić

Reply:
Z tego co się orientuje, to `null` ma taki sam efekt co pusty string